lib/msf/core/modules/loader/base.rb

```diff
@@ -466,9 +466,17 @@ def module_path?(path)
 
   # Tries to determine if a file might be executable,
   def script_path?(path)
-    File.executable?(path) &&
-      !File.directory?(path) &&
-      ['#!', '//'].include?(File.read(path, 2))
+    # warn users if their external modules aren't marked executable
+    # per #14281
+    if File.directory?(path) || !['#!', '//'].include?(File.read(path, 2))
+      false
```

Comment:
Won't this condition still be silently failing? Shouldn't this be updated to log some info here?

Reply:
After review this again this seems like a real mess to fix since `path` will contain every file underneath the Metasploit root path and could contain files with different file starts that may still be valid. If we start throwing errors at this point in the code we are just going to clutter the log file. I'll leave this for another PR to fix if and when they want to fix it.